tests/test_installables.py

```
@@ -0,0 +1,160 @@
+            mock_run.side_effect = [
+                CompletedProcess(args=[], returncode=0, stdout=stdout, stderr=""),
+                CompletedProcess(args=[], returncode=0, stdout="", stderr=""),
+            ]
+            result, message = git.ensure_submodules_state(Path("/repo"))
+
+        assert result is True
+        assert message == ""
+        assert mock_run.call_args_list[1].args[0] == expected_command
+
+    @pytest.mark.parametrize("init_submodules", [True, False])
+    def test_ensure_submodules_state_noop_when_matching(self, git: GitRepo, init_submodules: bool):
```

Ruff (0.15.10)

[warning] Boolean-typed positional argument in function definition

(FBT001)